### DISCUSSION TITLE
Most performant way to draw a line, outline

### DISCUSSION THREAD

**Discussion by k11q** [pixijs/pixijs#9007] [Q&A] [answered]
Hi I am new in pixi and webgl, or any sort of 2d/3d. I see there are many methods to create a line. Which one is the most performant way and best way?

A side question, is there another method for doing outline other than the plugin? I am using the plugin and it's not as sharp as I need it to be.

Thank you.

**Comment by miltoncandelero** [comment, marked as answer]
The sharpest line you will get is with [pixi-smooth-graphics](https://github.com/pixijs/graphics-smooth/)

The one without plugins is from `Graphics`

The most performant way to draw a line would be to use an sprite :P

Outlines have nothing to do with lines, there is an Outline Filter that can be expensive. Cheapest way to do outlines would be with a second sprite behind the first one 😅

**Reply by k11q** [reply]
Thanks, how may I put a sprite behind, considering I want to add it dynamically to another sprite. Do I need to create a container and fill it with both sprite?

**Reply by miltoncandelero** [reply]
yep, a container with two sprites inside it
